### PR TITLE
Add Solr Quickstart

### PR DESCRIPTION
**Pull request by pritamdas99**
Signed-off-by: pritamdas99 <pritam@appscode.com>

### CODE REVIEW

File: docs/guides/solr/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+|------------------------------------|:------------:|
+| Clustering                         |   &#10003;   |
+| Authentication & Autorization      |   &#10003;   | 
+| Custom Configuration               |   &#10003;   | 
+| Grafana Dashboards                 |   &#10003;   | 
+| Externally manageable Auth Secret  |   &#10003;   |
+| Persistent Volume                  |   &#10003;   |
+| Builtin Prometheus Discovery       |   &#10003;   | 
+| Using Prometheus operator          |   &#10003;   |
+| Solr Builtin UI                    |   &#10003;   |
```
Comment (raihankhan) [inline docs/guides/solr/README.md]: Add/Update - 
- Reconfigurable Health Checker
- Dashboard ( Solr UI )
- Monitoring with Prometheus & Grafana
- Alert Dashboards

File: docs/guides/solr/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    weight: 10
+menu_name: docs_{{ .version }}
+section_menu_id: guides
+url: /docs/{{ .version }}/guides/solr/
+aliases:
+  - /docs/{{ .version }}/guides/solr/README/
+---
+
+> New to KubeDB? Please start [here](/docs/README.md).
+
```
Comment (raihankhan) [inline docs/guides/solr/README.md:17]: Add a overview block like this -

# Overview 

SingleStore, a distributed SQL database for real-time analytics, transactional workloads, and operational applications. With its in-memory processing and scalable architecture, SingleStore enables organizations to achieve high-performance and low-latency data processing across diverse data sets, making it ideal for modern data-intensive applications and analytical workflows.

File: docs/guides/solr/quickstart/overview/index.md [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+namespace/demo created
+
+$ kubectl get namespace
+NAME                 STATUS   AGE
+demo                 Active   9s
+```
+
+> Note: YAML files used in this tutorial are stored in [guides/solr/quickstart/overview/yamls](https://github.com/kubedb/docs/tree/{{< param "info.version" >}}/docs/guides/solr/quickstart/overview/yamls) folder in GitHub repository [kubedb/docs](https://github.com/kubedb/docs).
+
+> We have designed this tutorial to demonstrate a production setup of KubeDB managed Solr. If you just want to try out KubeDB, you can bypass some safety features following the tips [here](/docs/guides/solr/quickstart/overview/index.md#tips-for-testing).
```
Comment (raihankhan) [inline docs/guides/solr/quickstart/overview/index.md:42]: Fix links

File: docs/guides/solr/quickstart/overview/index.md [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+NAME     VERSION   DB_IMAGE                              DEPRECATED   AGE
+8.11.2   8.11.2    ghcr.io/appscode-images/solr:8.11.2                9d
+9.4.1    9.4.1     ghcr.io/appscode-images/solr:9.4.1                 9d
+```
+
+Notice the `DEPRECATED` column. Here, `true` means that this SolrVersion is deprecated for the current KubeDB version. KubeDB will not work for deprecated SolrVersion.
+
+In this tutorial, we will use `9.4.1` SolrVersion CR to create an Solr cluster.
+
+> Note: An image with a higher modification tag will have more features and fixes than an image with a lower modification tag. Hence, it is recommended to use ZooKeeperVersion CRD with the highest modification tag to take advantage of the latest features. For example, use `xpack-8.11.1` over `7.9.1-xpack`.
```
Comment (raihankhan) [inline docs/guides/solr/quickstart/overview/index.md]: remove xpack refs

File: docs/guides/solr/quickstart/overview/index.md [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+
+## Create an Solr Cluster
+
+The KubeDB operator implements a Solr CRD to define the specification of a Solr database.
+
+The Kubedb Solr runs in solrcloud mode. Hence, it needs a external zookeeper to distribute replicas among pods and save configurations.
```
Comment (raihankhan) [inline docs/guides/solr/quickstart/overview/index.md]: solrcloud > make it bold

**Review by raihankhan — CHANGES_REQUESTED** [review]
remove statefulset references and elasticsearch copy paste typos

**Review by obaydullahmhs — APPROVED** [review]
LGTM

**Review by raihankhan — APPROVED** [review]

**Pull request merged**